File: src/main/java/my/study/repository/design/pattern/builder/java/UseDemo.java
package my.study.repository.design.pattern.builder.java;

/**
 * Builder design mode, when the object is complex, need a lot of construct method
 * you can use this to simple construct a complex object.
 *
 * @author Djh on 2018/7/27 15:36
 * E-Mail ：1544579459@qq.com
 */
public class UseDemo {
    public static void main(String[] args) {
        ComplexObject.Builder builder = new ComplexObject.Builder();
        ComplexObject complexObject = builder.setAge(22)
                .setFatherName("djh")
                .setSex("boy")
                .setName("djl")
                .build();
    }
}

File: src/main/java/my/study/repository/design/pattern/builder/kotlin/UseDemo.kt
package my.study.repository.design.pattern.builder.kotlin

/**
 * @author djh on  2018/10/3 20:32
 * @E-Mail 1544579459@qq.com
 */
object UseDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        val builder = ComplexObject.Builder().apply {
            setAge(21)
            setSex("boy")
            setName("djh")
            setFatherName("dxb")
        }
        val complexObject = builder.build()
        println(complexObject.mAge)
    }
}